linkerd/app/core/src/errors/respond.rs

```diff
@@ -32,10 +32,11 @@ pub trait HttpRescue<E> {
 
 #[derive(Clone, Debug)]
 pub struct SyntheticHttpResponse {
-    pub grpc_status: tonic::Code,
-    pub http_status: http::StatusCode,
-    pub close_connection: bool,
-    pub message: Cow<'static, str>,
+    grpc_status: tonic::Code,
+    http_status: http::StatusCode,
+    close_connection: bool,
+    message: Cow<'static, str>,
+    location: Option<http::header::HeaderValue>,
```

Comment:
tiny style nit: `HeaderValue` is imported:
```suggestion
    location: Option<HeaderValue>,
```